core/indexer/common.go

```diff
@@ -282,18 +304,35 @@ func prepareBufferMiniblocks(buff bytes.Buffer, meta, serializedData []byte) byt
 	return buff
 }
 
-func serializeBulkTxs(bulk []*Transaction, selfShardID uint32) bytes.Buffer {
+func serializeBulkTxs(
+	bulk []*Transaction,
+	selfShardID uint32,
+	foundedObjMap func(hashes []string, index string) (map[string]bool, error),
+) bytes.Buffer {
 	var buff bytes.Buffer
 	var err error
 
+	txsHashes := make([]string, len(bulk))
+	for idx := range bulk {
+		txsHashes[idx] = bulk[idx].Hash
+	}
+
+	existsInDb, err := foundedObjMap(txsHashes, txIndex)
+	if err != nil {
+		log.Warn("indexer", "error", err.Error())
+		return buff
+	}
+
 	for _, tx := range bulk {
 		var meta, serializedData []byte
-
-		if isCrossShardDstMe(tx, selfShardID) && tx.Status != txStatusInvalid {
-			// update tx
+		if existsInDb[tx.Hash] {
```

Comment:
maybe extract this if/else into a function like `getSerializedData`?

Reply:
done.